src/Services/PatientService.php

```diff
@@ -169,30 +129,46 @@ public function insert($data)
             $sql,
             $query['bind']
         );
+
         if ($results) {
-            return $fresh_pid;
+            $processingResult->setData($freshPid);
+            return $processingResult;
+        } else {
+            $processingResult->addProcessingError("error processing SQL Insert");
         }
 
-        return $results;
+        return $processingResult;
     }
 
+    /**
+     * Updates an existing patient record.
+     * 
+     * @param $pid - The patient identifier (PID) used for update.
+     * @param $data - The updated patient data fields
+     * @return
+     */
     public function update($pid, $data)
     {
-        $validationResult = $this->validate($data, 'update', $pid);
-        if ($validationResult->isNotValid()) {
-            return $validationResult;
+        $data["pid"] = $pid;
```

Comment:
we're including the `pid` in the data to be validated since that's part of the patient record definition.